ArduPlane/Attitude.cpp

```diff
@@ -5,31 +5,76 @@
   to PIDs to change the scaling of the PID with speed. At high speed
   we move the surfaces less, and at low speeds we move them more.
  */
-float Plane::calc_speed_scaler(void)
+Vector3f Plane::calc_rpy_speed_scaler(void)
 {
-    float aspeed, speed_scaler;
+    Vector3f speed_scaler;
+    float aspeed;
     if (ahrs.airspeed_estimate(aspeed)) {
         if (aspeed > auto_state.highest_airspeed && hal.util->get_soft_armed()) {
             auto_state.highest_airspeed = aspeed;
         }
+
+        Vector3f aspeed_rpy;
+        if (is_positive(g2.prop_disc_loading)) {
+            // Calculate prop-wash exit velocity using momentum disc theory
+            const float rho = SSL_AIR_DENSITY * plane.barometer.get_air_density_ratio();
+            // Estimate propeller disc loading in N/m^2 assuming drag is negligible and X axis 
+            // specific force is mostly due to propeller thrust. This assumption will underestimate
+            // propeller disc loading at higher speeds but will be accurate enough at low speeds
+            // which is when propwash effects dominate.
+            const float disc_loading = MAX(g2.prop_disc_loading * ahrs.get_accel().x, 0.0f);
+            // Solve the quadratic for propeller inflow factor a^2 + a = T / (2 * rho * S * V^2)
+            // a = inflow factor
+            // T = propeller thrust (N)
+            // rho = air density (Kg/m^3)
+            // S = propeller area (m^2)
+            // V = true airspeed (m/s)
+            const float a_coef = 1.0f;
+            const float b_coef = 1.0f;
+            const float c_coef = - disc_loading / (2.0f * rho * sq(MAX(aspeed,0.1f)));
+            const float inflow_factor = (-b_coef + sqrtf(sq(b_coef) - 4.0f * a_coef * c_coef)) / (2.0f * a_coef);
+            aspeed_rpy.x = aspeed * (1.0f + 2.0f * inflow_factor * g2.propwash_roll_comp);
+            aspeed_rpy.y = aspeed * (1.0f + 2.0f * inflow_factor * g2.propwash_pitch_comp);
+            aspeed_rpy.z = aspeed * (1.0f + 2.0f * inflow_factor * g2.propwash_yaw_comp);
+        } else {
+            aspeed_rpy.x = aspeed_rpy.y = aspeed_rpy.z = aspeed;
```

Comment:
we could add a set method or equals over load to set all values to a single float.

Reply:
This PR only does it on one place. If we are going to add a method, then to make it worth while and realise a saving in flash space, it should be done in a separate PR and applied to all places on code where this operation is being done.

Reply:
There are three, here and lines 92 and 95. I find this multi equals harder to read, I think that is more of a issue than the flash saving. But as you say it not a blocker for this PR.